examples/opensora_hpcai/opensora/pipelines/train_pipeline.py

```diff
@@ -438,3 +446,91 @@ def unpatchify(self, x: Tensor):
         x = ops.transpose(x, (0, 4, 1, 2, 5, 3, 6))
         x = ops.reshape(x, (n, self.c, f, self.nh * self.p[1], self.nw * self.p[2]))
         return x
+
+
+class DiffusionWithLossPynative(DiffusionWithLoss):
```

Comment:
Do not overwrite the entire class just because of a tiny `with _no_grad()` change. Instead, create a custom context manager, i.e.:
```python
class NoGrad(_no_grad):
    def __init__(self):
        super().__init__()
        self._pynative = ms.get_context("mode") == ms.PYNATIVE_MODE

    def __enter__(self):
        if self._pynative:
            super().__enter__()

    def __exit__(self, *args):
        if self._pynative:
            super().__exit__(*args)
```
and then use `with NoGrad():` in both PYNATIVE and GRAPH modes.

Reply:
![image](https://github.com/mindspore-lab/mindone/assets/65380679/2100198c-657c-4104-9055-de1edcdeb937)
Have tried it, and got this error

Reply:
Try wrapping it in [`jit_class`](https://www.mindspore.cn/docs/en/r2.3.0rc2/api_python/mindspore/mindspore.jit_class.html)?

Reply:
I think you can write the “witn no_grad” out of the scope of construct, avoids copy and paste the entire codes twice,